featuretools/tests/primitive_tests/test_feature_serialization.py

```diff
@@ -14,16 +15,27 @@ def test_pickle_features(es):
     filepath = os.path.join(dir_path, 'test_feature')
 
     ft.save_features(features_original, filepath)
-    features_deserialized = ft.load_features(filepath)
-    for feat_1, feat_2 in zip(features_original, features_deserialized):
-        assert feat_1.unique_name() == feat_2.unique_name()
-        assert feat_1.entityset == feat_2.entityset
-
-    # file is smaller than entityset in memory
+    features_deserializedA = ft.load_features(filepath)
     assert os.path.getsize(filepath) < asizeof(es)
+    os.remove(filepath)
 
+    with open(filepath, "w") as f:
+        ft.save_features(features_original, f)
+        f.close()
+    features_deserializedB = ft.load_features(open(filepath))
+    assert os.path.getsize(filepath) < asizeof(es)
     os.remove(filepath)
 
+    features = ft.save_features(features_original)
+    features_deserializedC = ft.load_features(features)
+    assert asizeof(features) < asizeof(es)
+
+    features_deserialized_options = [features_deserializedA, features_deserializedB, features_deserializedC]
+    for features_deserialized in features_deserialized_options:
+        for feat_1, feat_2 in zip(features_original, features_deserialized):
+            assert feat_1.unique_name() == feat_2.unique_name()
+            assert feat_1.entityset == feat_2.entityset
+
 
 def test_pickle_features_with_custom_primitive(es):
```

Comment:
I think we can simplify these two tests since they are basically the same tests and assertions but with different features.  We can make a helper function that runs all the assertions and takes a feature list and entityset as input, then all the tests have to do is generate the features and call the helper function